tests/testPreallocator.py

```diff
@@ -1,51 +1,156 @@
 import unittest
+import random
 
 from preallocator import *
 
 from config import Config
-
+from tangoObjects import TangoMachine
 
 class TestPreallocator(unittest.TestCase):
 
     def setUp(self):
-        return False
+        if Config.VMMS_NAME == "tashiSSH":
+            from vmms.tashiSSH import TashiSSH
+            vmms = TashiSSH()
+            self.preallocator = Preallocator({"tashiSSH": vmms})
+        
+        elif Config.VMMS_NAME == "ec2SSH":
+            from vmms.ec2SSH import Ec2SSH
+            vmms = Ec2SSH()
+            self.preallocator = Preallocator({"ec2SSH": vmms})
+
+        elif Config.VMMS_NAME == "localDocker":
+            from vmms.localDocker import LocalDocker
+            vmms = LocalDocker()
+            self.preallocator = Preallocator({"localDocker": vmms})
+
+        elif Config.VMMS_NAME == "distDocker":
+            from vmms.distDocker import DistDocker
+            vmms = DistDocker()
+            self.preallocator = Preallocator({"distDocker": vmms})
+
+        else:
+            vmms = None
+            self.preallocator = Preallocator({"default": vmms})
+       
+        self.vm = TangoMachine()
```

Comment:
I think many of the errors are stemming from the fact that `vmms` was not passed into the TangoMachine here, but there are still failures after this change is made

Reply:
Hi @fanpu. Thank you for this information! I interpreted something else as a successful run of these tests. I will look into the issues with this code.

Reply:
Please let me know if you run into any issues with the setup process for testing, https://docs.autolabproject.com/tango/ is a good reference